keps/sig-cli/3659-kubectl-apply-prune/README.md

```diff
@@ -627,128 +544,171 @@ In future, we may define additional “index” mechanisms here to further optim
-the objects before applying (with a set of label-filtered LIST requests),
-we will explore to what extent that can be amortized over other kubectl
-operations in alpha.  One interesting option may be to use the fieldManager,
-choosing a fieldManager that includes the ApplySet ID to automatically
-detect conflicts (by _not_ specifying force); we intend to explore
-how this looks in practice and whether other options present themselves.
+The `--namespace` flag will be required when using any namespaced parent, including the default Secret. Because that flag throws a mismatch error when the set contains resources with heterogeneous namespaces, this limits the scope of ApplySet-based pruning in kubectl specifically beyond what the spec proposed strictly requires. Specifically, in kubectl, ApplySets spanning multiple namespaces MUST use a cluster-scoped parent object. We believe this limitation is reasonable and encourages best practices, but we could consider relaxing this position (e.g. using the ApplySet-in-input option described above) based on user feedback. When applicable, kubectl will ensure that all "visited" namespaces (as defined in the current operational code) are named by the sum of the parent's own namespace (if any) and the `applyset.kubernetes.io/additional-namespaces` annotation.
+
+In the initial implementation, the initial apply operation will overwrite the `applyset.kubernetes.io/part-of` annotation (unless this is automatically prevented, e.g. by a conflict caused by SSA use on that label by two different tools). This effectively allows the set to thrash, which is undesirable. Before beta, we will explore whether it is possible to detect such "overlapping" ApplySets with acceptable performance, particularly in conjunction with server-side-apply, which does not require an object read before applying.
+The two most readily apparent options both increase requests made by O(N). The first is to fetch the objects before applying (which may not share any particular labels a priori); the second is to do an additional SSA on each object using the applyset field manager prior to the main operation (instead of injecting our labels into the data used for that main operation, which may be a client-side apply).
 
 We differentiate between "adoption" (taking over management of a set of
 objects created by another tool), vs "migration" (taking over management of
 a set of objects created with the existing pruning mechanism).
 
 We will not support "adoption" of existing ApplySets initially, other than
-by re-applying "over the top".  Based on user feedback, we may require a flag
-to adopt existing objects / ApplySets.
-
-In the alpha scope, we will explore suitable "migration" tooling for moving
-from existing `--prune` objects.  Note that migration is not trivial, in that
-different users may expect different behaviors with regard to the GKs selected
-or the treatment of objects having/lacking the `last-application-configuration`
```

Comment:
I'm not sure what we meant here. The migration command will be subject to the same gotchas in set selection as the old one, but I think all we can do is ask for the same inputs they used on their own command, and go with the results. We could add interactive confirmation maybe?